packages/react-core/src/modules/in-app-message/hooks/useInAppMessages.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  FetchInAppMessagesOptions,
+  InAppMessage,
+  InAppMessageClient,
+} from "@knocklabs/client";
+import { useStore } from "@tanstack/react-store";
+import { useEffect, useMemo } from "react";
+
+import { useInAppMessageChannel } from "../context";
+
+// TODO: Type so that the message content (or fields) is typed
+export const useInAppMessages = (
+  messageType: string,
+  options: FetchInAppMessagesOptions = {},
+) => {
+  const { inAppChannelClient } = useInAppMessageChannel();
+
+  const inAppMessageClient = useMemo(() => {
+    return new InAppMessageClient(inAppChannelClient, messageType, options);
+  }, [inAppChannelClient, messageType, options]);
+
+  // TODO: Is it possible to extract all store usage?
+  // Maybe not because framework specific hooks need to be a called to support reactivity
+  // Could definitely create selectors as functions in the client library though
+  const messages = useStore(inAppChannelClient.store, (state) => {
+    const messageIds = new Set(
```

Comment:
I'm feeling somewhat confused what's happening inside this useStore callback func. It looks like we are looping over the entire state.messages, but is that necessary? Shouldn't we get message ids from a cached response if available, and then look them up in state.messages?

Reply:
Yeah, the `reduce` is unnecessary. I'll update it to loop over the cached message ids and just grab the ones we need